### ISSUE TITLE
Regression in 4.2.5 re: decimal precision

### ISSUE THREAD

**Issue by jaredbeck** [rails/rails#22325]
The following test passes in 4.2.4, fails in 4.2.5.

This issue can be reproduced in sqlite, mysql, and postgres.

Ahh the heartbreak of floating point math :sob:

Thanks for having such an easy-to-use bug reporting template, though! :heart:

```ruby
begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  # Activate the gem you are reporting the issue against.
  gem 'activerecord', '4.2.5'
  gem 'sqlite3'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'

# Ensure backward compatibility with Minitest 4
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
    t.decimal :ukiyo_e, :precision => 6, :scale => 4
  end
end

class Post < ActiveRecord::Base
end

class BugTest < Minitest::Test
  def test_the_floating_world
    decimal = 0.12345
    post = Post.create!(ukiyo_e: decimal)
    assert_in_delta post.reload.ukiyo_e, decimal, 0.00001
  end
end
```

The above test produces the following output:

```
  1) Failure:
BugTest#test_the_floating_world [active_record_gem.rb:39]:
Expected |0.1235 - 0.12345| (0.00005) to be <= 1.0e-05.
```

**Comment by rafaelfranca** [comment]
That behavior is correct. If you have a scale of 4 you want only 4 digits in the fractional part. This is why Rails is rounding it to 4 digits in the fractional part. See https://github.com/rails/rails/issues/22006#issuecomment-149686388

**Comment by jaredbeck** [comment]
So, 4.2.4 does not "respect" scale, and that was fixed in 4.2.5 by https://github.com/rails/rails/commit/0b94384bef44535a408b034e44b90d08318e258a? Thanks! Thanks for the quick response.

**Comment by rafaelfranca** [comment]
Yes, exactly that. It before 4.2.5 scale was not Ben being respected
